doc/sphinx/source/developer_guide/modules.rst

```diff
@@ -4,6 +4,535 @@
+|               +---------------+-------------------------------------------------------------------------+
+|               | dl_setting    | Setting description, see :ref:`Settings <label-defines-and-settings>`   |
+|               | (child of     | section for details                                                     |
+|               | dl_settings)  |                                                                         |
+|               |               |                                                                         |
+|               |               |                                                                         |
++---------------+---------------+-------------------------------------------------------------------------+
+|               |               | Comma separated list of required modules                                |
+| | dep         |               |                                                                         |
+| | (0 or 1)    | depends       |                                                                         |
+|               |               | Allows to specify OR dependencies with pipe                             |
+|               |               | (\|) similar to Debian depends, ex: module1,module2|module3             |
+|               |               | would make it depend on | module1 AND (module2 OR module3)              |
+|               |               |                                                                         |
+|               |               | The elements can be a module name (as set in the module XML ``name``    |
+|               |               | node) or a functionality, which has to be preceded by @                 |
+|               +---------------+-------------------------------------------------------------------------+
+|               | provides      | Advertises the functionality that the module provides (e.g. actuators,  |
+|               |               | imu)                                                                    |
+|               +---------------+-------------------------------------------------------------------------+
+|               | conflicts      | Comma separated list of conflicting modules                            |
+|               |               |                                                                         |
+|               |               | The elements can be a module name (as set in the module XML ``name``    |
+|               |               | node) or a functionality, which has to be preceded by @                 |
++---------------+---------------+-------------------------------------------------------------------------+
+| | autoload    | name          | The name of the module which should also be automatically loaded        |
+| | (0 or 1)    |               |                                                                         |
```

Comment:
Not exactly. At the moment we don't have a nice way to load alternate modules like gps or imu for simulation. So you can't always skip this autoload thing (it will try to load them even if the parent module is not selected). But in general, for "normal" dependency it should not be used.